src/io/sampling.jl

```diff
@@ -503,73 +504,49 @@ Get factor default_bounds. Refer to `get_bounds` for more details of how the bou
 function get_default_bounds(dom::Domain, factor::Symbol)::Tuple
     model::Model = dom.model
     factor_filter::BitVector = collect(model[:fieldname]) .== factor
+    default_bounds::Tuple = model[:default_bounds][factor_filter][1]
 
-    default_bounds = if _is_discrete_factor(dom, factor)
-        default_lower, default_upper = model[:default_bounds][factor_filter][1]
-        (default_lower, default_upper - 1.0)
-    else
-        model[:default_bounds][factor_filter][1]
-    end
-
+    _is_discrete_factor(dom, factor) && return (default_bounds[1], default_bounds[2] - 1.0)
     return default_bounds
 end
 
 """
     set_factor_bounds!(dom::Domain, factor::Symbol, new_bounds::Tuple)::Nothing
     set_factor_bounds!(dom::Domain; factors...)::Nothing
 
-Set new lower and upper bounds for a parameter, `new_bounds = (lower_bound, upper_bound)`.
-All sampled values `s_vals` for this parameter will lie within the range
-`lower_bound ≤ s_vals ≤ upper_bound`. When used to set new bounds for a discrete parameter,
-the upper bound will be set to `upper_bound + 1` to guarantee that the sample values will
-always be less or equal `upper_bound`.
+Set new lower and upper bounds for a parameter. All sampled values `s_vals` for this
+parameter will lie within the range `lower_bound ≤ s_vals ≤ upper_bound`. When used to set
```

Comment:
"Sampled values for a parameter will lie within the range `lower_bound ≤ s_vals ≤ upper_bound`, where `s_vals` is the value of the factor"

... or something like the above.